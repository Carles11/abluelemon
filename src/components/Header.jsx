// @flow

import React from 'react'
import Navigation from './Navigation'

const nav = [
  { name: 'Home', url: '/' },
  { name: 'About us', url: '/about-us' },
  { name: 'Projects', url: '/projects' },
  { name: 'Contact', url: '/contact' },
]

type State = {
  showMenu: boolean
}

class Header extends React.Component<*, State> {
  state: State = {
    showMenu: false
  }

  handleClick = () => {
    this.setState(prevState => ({ showMenu: !prevState.showMenu }))
  }

  render() {
    const { showMenu } = this.state
    const classBtn = showMenu ? 'icon-cross' : 'icon-menu'

    return (
      <div className='app-header'>
        <button 
          onClick={this.handleClick}
          className={`btn ${classBtn}`} 
        />
        {showMenu && <Navigation items={nav} />}
      </div>
    )
  }
}

export default Header