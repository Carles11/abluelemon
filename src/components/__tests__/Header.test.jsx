// @flow

import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('<Header />', () => {
  it('should render itself', () => {
    const component = shallow(<Header />)
    expect(component).toMatchSnapshot()
  })

  it('should render the Nav Menu when the button is clicked', () => {
    const component = shallow(<Header />)
    expect(component.find('Navigation').length).toEqual(0)
    component.find('button').simulate('click')
    expect(component.find('Navigation').length).toEqual(1)
  })
})