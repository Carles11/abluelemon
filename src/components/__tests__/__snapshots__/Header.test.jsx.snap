// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn icon-menu"
          onClick={[Function]}
/>,
        false,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "btn icon-menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn icon-menu"
            onClick={[Function]}
/>,
          false,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "btn icon-menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
